proto/src/main/protobuf/bosatsu/TypedAst.proto

```diff
@@ -327,3 +327,56 @@ message Package {
 message Packages {
   repeated Package packages = 1;
 }
+
+// Message representing a Version using Semantic Versioning
+message Version {
+  int32 major = 1;
+  int32 minor = 2;
+  int32 patch = 3;
+  string pre_release = 4;
+  string build = 5;
+}
+
+// Message representing a Descriptor for versioning and identification
+message LibDescriptor {
+  Version version = 1;          // Semantic versioning information
+  int32 depth = 2;              // Depth in the dependency graph
+  string vcs_indent = 3;        // Version control system identifier (e.g., git:commit:aae)
+  repeated string hashes = 4;   // List of relevant hashes for verification
+  repeated string uris = 5;     // URIs where the dependency can be obtained
+}
+
+// Message representing the history of versions for a Library or Dependency
+message LibHistory {
+  LibDescriptor previous_patch = 1;  // Descriptor for the previous patch version, if any
+  LibDescriptor previous_minor = 2;  // Descriptor for the previous minor version, if any
+  LibDescriptor previous_major = 3;  // Descriptor for the previous major version, if any
+}
```

Comment:
Add some_previous which is a repeated random link uniform on `r`with distance proportional to `2^r` (to make it navigable quickly) (maybe around log depth previous links).